src/storage/src/manifest/storage.rs

```diff
@@ -94,13 +98,15 @@ impl LogIterator for ObjectStoreLogIterator {
 #[derive(Clone, Debug)]
 pub struct ManifestObjectStore {
     object_store: ObjectStore,
+    layer: Arc<ComposeLayer>,
     path: String,
 }
 
 impl ManifestObjectStore {
     pub fn new(path: &str, object_store: ObjectStore) -> Self {
         Self {
             object_store,
+            layer: Arc::new(ComposeLayer::new()),
```

Comment:
Will it break backward compatibility? If a user's old manifest wasn't being compressed.